### PR TITLE
Refactor covid data

### PR DESCRIPTION
**Pull request by johnschoeman**
Why:
We would like to update data widget being shown on the covid data card
on the home screen. Currently a lot of the logic in the CovidData
context and feature is implicitly dependent on the format of the data
being pre calculated to a set of trends. In order to better support
changes in this feature, We would like to:

- More fully separate the view layer calculation of aggregates and the
raw data itself.

- More fully separate the parsing of network data and the applications
internal concept of a set of covid data.

This commit:
Is a refactor of the covid data analytics widget feature. No new
functionality has been added.

Screen Shot 2020-10-28 at 10 55 34 AM

### CODE REVIEW

File: src/CovidData/Card/index.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React, { FunctionComponent } from "react"
+
+import { useCovidDataContext } from "../Context"
+import CovidDataCard from "./CovidDataCard"
+
+const CovidDataCardConnector: FunctionComponent = () => {
+  const { request: dataRequest, locationName } = useCovidDataContext()
```
Comment (devinjameson) [inline src/CovidData/Card/index.tsx]:
```suggestion
  const { request, locationName } = useCovidDataContext()
```

File: src/CovidData/Card/index.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  return <CovidDataCard dataRequest={dataRequest} locationName={locationName} />
```
Comment (devinjameson) [inline src/CovidData/Card/index.tsx]:
```suggestion
  return <CovidDataCard dataRequest={request} locationName={locationName} />
```

File: src/CovidData/PercentageChart.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import React, { ReactNode, FunctionComponent } from "react"
+import { StyleSheet, View } from "react-native"
+import { SvgXml } from "react-native-svg"
```
Comment (devinjameson) [inline src/CovidData/PercentageChart.tsx:3]: Maybe we can organize these imports like the rest?

File: src/CovidData/Context.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  request: initialRequest,
+}
+
+export const CovidDataContext = createContext<CovidDataContextState>(
+  initialState,
+)
+
+export const CovidDataContextProvider: FunctionComponent = ({ children }) => {
+  const { stateAbbreviation, displayCovidData } = useConfigurationContext()
+  const [request, setCovidDataRequest] = useState(initialRequest)
```
Comment (devinjameson) [inline src/CovidData/Context.tsx]: Is there a reason this doesn't use the conventional naming scheme?

File: src/CovidData/covidDataAPI.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      date,
+      peoplePositiveNewCasesCt,
+      peoplePositiveCasesCt,
+      peopleDeathCt,
+      peopleDeathNewCt,
+    } = networkDatum
+
+    return {
+      date,
+      positiveCasesNew: peoplePositiveNewCasesCt,
```
Comment (devinjameson) [inline src/CovidData/covidDataAPI.ts]: Thoughts on swapping this so the assignment happens in the destructuring from `networkDatum`?

**Review by devinjameson — APPROVED** [review]

File: src/CovidData/covidData.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+export type CovidDatum = {
+  date: Date
+  positiveCasesTotal: number
+  positiveCasesNew: number
+  deathTotal: number
+  deathNew: number
+}
+
+export const valid = (_data: CovidData): boolean => true
```
Comment (johnschoeman) [inline src/CovidData/covidData.ts]: This should probably be removed

File: src/Home/SectionButton.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import React, { FunctionComponent } from "react"
+import { View, StyleSheet } from "react-native"
+import { SvgXml } from "react-native-svg"
+
+import { Text } from "../components"
```
Comment (devinjameson) [inline src/Home/SectionButton.tsx:5]:
```suggestion
import { Text } from "../components"

import { Icons } from "../assets"
import { Typography, Buttons, Iconography, Colors } from "../styles"
```

File: src/CovidData/Context.tsx (hunk 4 above)
Comment (johnschoeman) [reply src/CovidData/Context.tsx]: nope, can fix!

**Pull request merged**